index.js

```diff
@@ -74,6 +74,17 @@ function walkdir() {
 
       if (dirent.isDirectory()) {
         queue.push(nextpath);
+      } else if (dirent.isSymbolicLink()) {
+        // If it's a symlink, check if the symlink points to a directory
+        fs.stat(nextpath, function (err, stats) {
+          if (err) {
+            return cb(err);
+          }
```

Comment:
I'm not really sure what to do about the error handling here. This `cb` tells the queue an error occured, but presumably `dirents.forEach(processDirent);` will continue processing, which could potentially error again...

Should it instead just throw the error and hope queue catches and emits it properly?

Reply:
Thanks for pointing this out! It has me rethinking how things are queued, but I'll make those changes after I land this